Forta-Bots-Challenge-3/src/agent.spec.ts

```diff
@@ -0,0 +1,315 @@
+    handleBlock = provideHandleBlock(mockProvider as any, mockContractAddresses);
+    const arbitrumFindings = await handleBlock(mockBlockEvent);
+
+    expect(optimismFindings.concat(arbitrumFindings)).toStrictEqual([
+      expect.objectContaining(
+        createTestFinding(mockOptimismChainId, "Optimism", [BigNumber.from(10), BigNumber.from(9)])
+      ),
+    ]);
+  });
+
+  it("returns correct findings if L2 supply invariant is violated on Arbitrum", async () => {
+    mockProvider = resetProvider(mockEthereumChainId);
+    mockEscrowBalances(
+      mockProvider,
+      mockBlockEvent.blockNumber,
+      mockContractAddresses.l1DaiAddress,
+      [mockContractAddresses.optimismEscrow, mockContractAddresses.arbitrumEscrow],
+      [10, 10]
+    );
+    handleBlock = provideHandleBlock(mockProvider as any, mockContractAddresses);
+    await handleBlock(mockBlockEvent);
+
+    mockProvider = resetProvider(mockOptimismChainId);
+    mockL2Supply(mockProvider, mockBlockEvent.blockNumber, mockContractAddresses.l2DaiAddress, 10);
+    handleBlock = provideHandleBlock(mockProvider as any, mockContractAddresses);
+    const optimismFindings = await handleBlock(mockBlockEvent);
+
+    mockProvider = resetProvider(mockArbitrumChainId);
+    mockL2Supply(mockProvider, mockBlockEvent.blockNumber, mockContractAddresses.l2DaiAddress, 9);
+    handleBlock = provideHandleBlock(mockProvider as any, mockContractAddresses);
+    const arbitrumFindings = await handleBlock(mockBlockEvent);
+
+    expect(optimismFindings.concat(arbitrumFindings)).toStrictEqual([
+      expect.objectContaining(
+        createTestFinding(mockArbitrumChainId, "Arbitrum", [BigNumber.from(10), BigNumber.from(9)])
+      ),
+    ]);
+  });
+
+  it("returns correct findings if both L2 supply invariants are violated", async () => {
+    mockProvider = resetProvider(mockEthereumChainId);
+    mockEscrowBalances(
+      mockProvider,
+      mockBlockEvent.blockNumber,
+      mockContractAddresses.l1DaiAddress,
+      [mockContractAddresses.optimismEscrow, mockContractAddresses.arbitrumEscrow],
+      [10, 10]
+    );
+    handleBlock = provideHandleBlock(mockProvider as any, mockContractAddresses);
+    await handleBlock(mockBlockEvent);
+
+    mockProvider = resetProvider(mockOptimismChainId);
+    mockL2Supply(mockProvider, mockBlockEvent.blockNumber, mockContractAddresses.l2DaiAddress, 9);
+    handleBlock = provideHandleBlock(mockProvider as any, mockContractAddresses);
+    const optimismFindings = await handleBlock(mockBlockEvent);
+
+    mockProvider = resetProvider(mockArbitrumChainId);
+    mockL2Supply(mockProvider, mockBlockEvent.blockNumber, mockContractAddresses.l2DaiAddress, 9);
+    handleBlock = provideHandleBlock(mockProvider as any, mockContractAddresses);
+    const arbitrumFindings = await handleBlock(mockBlockEvent);
+    expect(optimismFindings.concat(arbitrumFindings)).toStrictEqual([
```

Comment:
As a good practice, we don't concat findings because it can lead you to believe something is correct if it isn't.

For example, if `optimismFindings` were to have zero findings, and `arbitrumFindings` were to have the two expected findings, in the end, you'd end up with both expected findings, but they wouldn't have been generated as you expected.

So instead, we would check "as we go". So check `optimismFindings` at its point of creation, and same for `arbitrumFindings`.